### DISCUSSION TITLE
Datadog support

### DISCUSSION THREAD

**Discussion by kirillzh** [touchlab/CrashKiOS#89] [Feature request]
We use Datadog for crash reports, would love to migrate over to Kermit for that, but it doesn't currently support Datadog. Wondering if there is any existing work around that, or what's an estimated amount effort to add the support?

**Comment by alexfanatics** [comment]
We looked into this for a bit too. I think the main issue with Datadog compatibility is symbolication of exception stack traces. 
> Symbolication isn't yet supported for other kind of errors (log errors, span errors nor manually collected RUM errors)

https://github.com/DataDog/dd-sdk-ios/issues/1572#issuecomment-1851924128

Crashlytics + Bugsnag have an easy API to log an Exception along with the stacktrace which will be symbolicated on the web app, but it seems that Datadog does not support this.

```kotlin
// firebase
val exceptionClassName = throwable::class.qualifiedName
val exModel = FIRExceptionModel.exceptionModelWithName(exceptionClassName, throwable.message)!!
exModel.setStackTrace(throwable.getFilteredStackTraceAddresses().map { FIRStackFrame.stackFrameWithAddress(it.convert()) })
FIRCrashlytics.crashlytics().recordExceptionModel(exModel)

// bugsnag
val exception = throwable.asNSException()
val causes = throwable.causes.map { it.asNSException() }
// Notify will persist unhandled events, so we can safely terminate afterwards.
// https://github.com/bugsnag/bugsnag-cocoa/blob/6bcd46f5f8dc06ac26537875d501f02b27d219a9/Bugsnag/Client/BugsnagClient.m#L744
Bugsnag.notify(exception) { event ->
    if (event == null) return@notify true

    if (handled) {
        event.severity = BSGSeverity.BSGSeverityWarning
    } else {
        event.unhandled = true
        event.severity = BSGSeverity.BSGSeverityError
    }

    if (causes.isNotEmpty()) {
        event.errors += causes.map { it.asBugsnagError() }
    }

    true
}
```
